### PR TITLE
[ISPN-7207] Cache creation requires specific permissions when using security manager

### PR DESCRIPTION
**Pull request by iweiss**
Issue: https://issues.jboss.org/browse/ISPN-7207

### CODE REVIEW

**Comment by iweiss** [comment]
This is the initial implementation. I'm looking for feedback before I continue.

File: core/src/main/java/org/infinispan/jmx/SecurityActions.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        doPrivileged((PrivilegedAction<Void>) () -> {
+            try {
+                mBeanServer.unregisterMBean(objectName);
+            } catch (Exception e) {
+            }
+            return null;
+        });
+    }
+
+    static Set<ObjectName> queryNames(ObjectName target, QueryExp query, MBeanServer mBeanServer) throws MalformedObjectNameException {
```
Comment (tristantarrant) [inline core/src/main/java/org/infinispan/jmx/SecurityActions.java]: Rename to queryMBeanNames

File: jcache/commons/src/main/java/org/infinispan/jcache/SecurityActions.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+/**
+ * SecurityActions for the org.infinispan.jcache package.
+ * <p>
+ * Do not move. Do not change class and method visibility to avoid being called from other
+ * {@link java.security.CodeSource}s, thus granting privilege escalation to external code.
+ *
+ * @author Ingo Weiss
+ * @since 9.0
+ */
+final class SecurityActions {
```
Comment (tristantarrant) [inline jcache/commons/src/main/java/org/infinispan/jcache/SecurityActions.java]: I'd also like to see a test in integrationtests/security-manager-it/ that tests mbean registration

Comment (iweiss) [inline jcache/commons/src/main/java/org/infinispan/jcache/SecurityActions.java]: Isn't ClusteredCacheManagerMBeanTest enough here?

Comment (iweiss) [inline jcache/commons/src/main/java/org/infinispan/jcache/SecurityActions.java]: Wouldn't it be suitable to use `ComponentsJmxRegistrationTest`?

File: core/src/main/java/org/infinispan/jmx/SecurityActions.java (hunk 1 above)
Comment (iweiss) [inline core/src/main/java/org/infinispan/jmx/SecurityActions.java]: Done

**Comment by tristantarrant** [comment]
Merged to master, 8.2.x and 8.1.x

**Pull request closed without merging**